### PR TITLE
Rfc30/blob

### PR DESCRIPTION
**Pull request by thomas-k-cameron**
## Motivation and Context
part of RFC30

## Description
Implements serde to `Blob` datatype as described in RFC30.

also refactors it to another file

## Testing
- checks if the result of serialized data matches the intended result
- checks if the result of de-serialized data matches the intended result
- checks if the result of serialized data can be de-serialized as intended
- checks the above three for both human-readable and non-human readable formats.

## Checklist
- [x] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Review by Velfi — APPROVED** [review]

**Comment by thomas-k-cameron** [comment]
Do you think you want to create a dedicated branch for temporary merging RFC30/*?

I'm pretty sure there will be some merge conflict when you do it.

**Comment by thomas-k-cameron** [comment]
I prefer having a dedicated branch on this repository.  

In addition to the merge conflict, it helps me keep track of my progress, write tests... etc

I could make a branch on my repo but I think it's less error prone to have it on this repository.

Would you mind creating a branch for me?

**Comment by jdisanti** [comment]
I created the [`unstable-serde-support`](https://github.com/awslabs/smithy-rs/tree/unstable-serde-support) branch that you can target these PRs to.

**Comment by thomas-k-cameron** [comment]
@Velfi @jdisanti 
Would you kindly merge my PR?

File: rust-runtime/aws-smithy-types/src/blob/mod.rs [hunk 1]
```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#[cfg(all(test, aws_sdk_unstable, feature = "serialize", feature = "deserialize"))]
+mod test;
```
Comment (Velfi) [inline rust-runtime/aws-smithy-types/src/blob/mod.rs:7]: Would you mind moving the tests into a module in this same file?

**Pull request merged**